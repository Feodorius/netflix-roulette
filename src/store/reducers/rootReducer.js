import initialState from "../initialState";
import { ACTION_FILTER, ACTION_SORT, ACTION_SEARCH, ACTION_GET_MOVIES } from "../../utils/constants";

const rootReducer = (state = initialState, { type, payload }) => {
    switch (type) {
        case ACTION_FILTER: {
            return { ...state, filterOption: payload };
        }
        case ACTION_SORT: {
            return { ...state, sortOption: payload };
        }
        case ACTION_SEARCH: {
            return { ...state, searchString: payload };
        }
        case ACTION_GET_MOVIES: {
            const { movies, totalMovies } = payload;               
            return { ...state, movies, totalMovies };
        }
        default: {
            return state;
        }
    }
};

export default rootReducer;