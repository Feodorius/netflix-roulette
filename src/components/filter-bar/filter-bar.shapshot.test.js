import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from "redux";
import rootReducer from "../../store/reducers/rootReducer";
import { BrowserRouter as Router } from "react-router-dom";

import FilterBar from '../filter-bar/filter-bar.component';

describe('Should render components properly:', () => {
    const store = createStore(rootReducer);

    test('render filter bar', () => {
        const { asFragment } = render(<Provider store={store}><Router><FilterBar /></Router></Provider>);
        expect(asFragment(<FilterBar />)).toMatchSnapshot();
    });
});