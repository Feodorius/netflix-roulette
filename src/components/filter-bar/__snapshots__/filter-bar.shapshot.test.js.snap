// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render components properly: render filter bar 1`] = `
<DocumentFragment>
  <div
    class="MuiTabs-root css-1ujnqem-MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      style="overflow: hidden; margin-bottom: 0px;"
    >
      <div
        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="0"
          type="button"
        >
          all
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          documentary
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          comedy
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          horror
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          crime
        </button>
      </div>
      <span
        class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
